### PR TITLE
ADEN-5278 Remove taboola & gcs

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/ADEN-5282

### CODE REVIEW

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 1]
```diff
@@ -219,37 +202,13 @@ public static function onWikiaSkinTopModules( &$scriptModules, $skin ) {
 	 * @return bool
 	 */
 	public static function onWikiaMobileAssetsPackages( array &$jsStaticPackages, array &$jsExtensionPackages, array &$scssPackages ) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php:204]: MAJOR Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
MAJOR Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

File: extensions/wikia/AdEngine/js/run/desktop.run.js [hunk 2]
```diff
@@ -11,14 +11,12 @@ require([
...
-	'ext.wikia.adEngine.taboolaHelper',
 	'ext.wikia.adEngine.slot.service.actionHandler',
 	'ext.wikia.adEngine.slotTracker',
 	'ext.wikia.adEngine.slotTweaker',
 	'ext.wikia.adEngine.sourcePointDetection',
 	'ext.wikia.aRecoveryEngine.adBlockDetection',
 	'wikia.window',
-	require.optional('ext.wikia.adEngine.recovery.gcs'),
 	require.optional('ext.wikia.adEngine.template.floatingRail')
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/run/desktop.run.js:21]: MAJOR Function has 17 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/context/adContext.js [hunk 3]
```diff
@@ -72,8 +67,7 @@ define('ext.wikia.adEngine.adContext', [
 	}
 
 	function updateAdContextRecoveryServices(context, noExternals) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/context/adContext.js:69]: MAJOR Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/config/desktop.js [hunk 4]
```diff
@@ -13,8 +13,7 @@ define('ext.wikia.adEngine.config.desktop', [
 	'ext.wikia.adEngine.provider.evolve2',
 	'ext.wikia.adEngine.provider.remnantGpt',
 	'ext.wikia.adEngine.provider.rubiconFastlane',
-	'ext.wikia.adEngine.provider.turtle',
-	require.optional('ext.wikia.adEngine.provider.taboola')
+	'ext.wikia.adEngine.provider.turtle'
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/config/desktop.js:17]: MAJOR Function has 11 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 89 issues
* ![MAJOR][MAJOR] 80 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2ContextService.class.php#L25](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L25): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2ContextService.class.php#L28](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L28): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L95](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L95): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L96](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L96): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L179](https://github.com/Wikia/app/blob/b54f76bb6aa77268ca8f6dc6aaf35e40a7067c61/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L179): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by kvas-damian — APPROVED** [review]

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**